bluemira/base/parameter_frame/_frame.py

```diff
@@ -64,10 +64,12 @@ def from_dict(
                 raise ValueError(f"Data for parameter '{member}' not found.")
 
             value_type = cls._validate_parameter_field(member)
-            kwargs[member] = NewParameter(**param_data, _value_types=value_type)
+            kwargs[member] = NewParameter(
+                name=member, **param_data, _value_types=value_type
+            )
```

Comment:
Possible extra addition allowing key value pair specification. We may not want to do this for other reasons (units, source etc) but just a thought
```suggestion
            if not isinstance(param_data, Dict):
                param_data = {"value": param_data}
            kwargs[member] = NewParameter(
                name=member, **param_data, _value_types=value_type
            )
```